AssemblerLib/LocalToGlobalIndexMap.cpp

```diff
@@ -74,52 +75,47 @@ LocalToGlobalIndexMap::LocalToGlobalIndexMap(
 }
 
 LocalToGlobalIndexMap::LocalToGlobalIndexMap(
-    std::vector<MeshLib::MeshSubsets*>&& mesh_subsets,
-    std::vector<std::size_t> const& original_indices,
+    std::vector<std::unique_ptr<MeshLib::MeshSubsets>>&& mesh_subsets,
+    std::size_t const component_id,
     std::vector<MeshLib::Element*> const& elements,
     AssemblerLib::MeshComponentMap&& mesh_component_map)
     : _mesh_subsets(std::move(mesh_subsets)),
       _mesh_component_map(std::move(mesh_component_map))
 {
-    assert(original_indices.size() == _mesh_subsets.size());
-    // For all MeshSubsets and each of their MeshSubset's and each element
-    // of that MeshSubset save a line of global indices.
+    // There is only on mesh_subsets in the vector _mesh_subsets.
```

Comment:
It seems to me that the original code was slightly more general than your solution is now. However, the old one was certainly not feature-complete. Since your PR is an intermediate step that's fine.